lib/protobuf/field/field_hash.rb

```diff
@@ -43,9 +43,13 @@ def merge!(other)
       alias update merge!
 
       # Return a hash-representation of the given values for this field type.
-      # The value in this case would be the hash itself.
+      # The value in this case would be the hash itself, right? Unfortunately
+      # not because the values of the map could be messages themselves that we
+      # need to transform.
```

Comment:
Interesting. I didn't realize this was expected to do a "deep" conversion. I figured whatever called it (JSON serialization?) would handle recursively converting values to hashes...